src/components/Superscript.js

```diff
@@ -34,6 +34,10 @@ const mapStateToProps = ({ contextViews, cursor, cursorBeforeEdit, modalData, sh
     ? (props.showContexts ? contextOf(cursor || []) : cursor || [])
     : thoughtsRanked
 
+  const notInArchive = context => (
+    showHiddenThoughts || context.context.indexOf('=archive') === -1
```

Comment:
Since `showHiddenThoughts` does not change over the course of the loop, it should be checked outside `filter` for better performance.